docs/fastdds/dds_layer/subscriber/dataReader/readingData.rst

```diff
@@ -6,36 +9,36 @@ Accessing received data
 The application can access and consume the data values received on the :ref:`dds_layer_subscriber_dataReader`
 by *reading* or *taking*.
 
- * **Reading** is done with the :func:`read_next_data` member function.
+ * **Reading** is done with the |DataReader::read_next_sample-api| member function.
    It reads the next, non-previously accessed data value available on the
-   :ref:`dds_layer_subscriber_dataReader`, and stores it in the provided data buffer.
- * **Taking** is done with the :func:`take_next_data` member function.
+   DataReader, and stores it in the provided data buffer.
+ * **Taking** is done with the |DataReader::take_next_sample-api| member function.
    It reads the next, non-previously accessed data value available on the
-   :ref:`dds_layer_subscriber_dataReader`, and stores it in the provided data buffer.
-   Additionally, it also removes the value from the :ref:`dds_layer_subscriber_dataReader`,
+   DataReader, and stores it in the provided data buffer.
+   Additionally, it also removes the value from the DataReader,
    so it is no longer accessible.
 
-If there is no unread data in the :ref:`dds_layer_subscriber_dataReader`, both operations will return
+If there is no unread data in the DataReader, both operations will return
 ``NO_DATA`` and nothing is returned.
 
 In addition to the data value, the data access operations also provide a :ref:`dds_layer_subscriber_sampleInfo`
 instance with additional information that help interpreting the returned data value, like the originating
 :ref:`dds_layer_publisher_dataWriter` or the publication time stamp.
-Please, refer to the :ref:`dds_layer_subscriber_sampleInfo` section for an extensive description of its contents.
+Please, refer to the SampleInfo section for an extensive description of its contents.
```

Comment:
As this is a 'please refer to...' sentence, I'd keep the link to the section. If we want to avoid having two links to the SampleInfo section on the same section, I'd remove the first one instead of this one.